ols/app/endpoints/ols.py

```diff
@@ -231,6 +231,42 @@ def validate_question(conversation_id: str, llm_request: LLMRequest) -> str:
         )
 
 
+def _validate_question_keyword(query: str) -> str:
+    """Validate user question using keyword."""
+    # Current implementation is without any tokenizer method, lemmatization/n-grams.
+    # Add valid keywords to keywords.txt file.
+    query_temp = query.lower()
+    for kw in config.keywords:
+        if kw in query_temp:
+            return constants.SUBJECT_VALID
+    # query_temp = {q_word.lower().strip(".?,") for q_word in query.split()}
+    # common_words = config.keywords.intersection(query_temp)
+    # if len(common_words) > 0:
+    #     return constants.SUBJECT_VALID
+
+    logger.info(f"No matching keyword found for query: {query}")
+    return constants.SUBJECT_INVALID
+
+
+def validate_question(conversation_id: str, llm_request: LLMRequest) -> str:
+    """Validate user question."""
+    if config.dev_config.disable_question_validation:
```

Comment:
1) i'm not sure we've made a decision on whether we'd allow an admin to disable validation.  i'm not sure there's a reason we wouldn't let them, if they choose to do it, though of course it should not be our default mode of operation
2) admins don't control the config.yaml directly, they have to edit the OLS CR to set configuration, so we don't have to allow "disabled" as a valid enum on the CRD, while we can still allow it in the config yaml.  (yes technically an admin could bypass the operator and modify the config directly, but then they could also set the dev_config flags via that mechanism.  At that point they are bypassing our official apis)

moving the "disable validation" mechanism into your new field keeps our config simpler/easier to reason about, rather than having 2 fields that contribute to the overall validation behavior.